### ISSUE TITLE
GPU Exclusive Application leaves CPU threads idle - Work Fetch does not get work

### ISSUE THREAD

**Issue by JacobWKlein** [BOINC/boinc#3679]
**Describe the bug**
When I have a GPU Exclusive Application setup, and that process is running, then Work Fetch does not get work for all of my CPUs.

**Steps To Reproduce**
1. Setup a GPU Exclusive Application. Mine are "FahCore_21.exe" and "FahCore_22.exe".
2. Make sure that you have work for your GPU(s), that is downloaded and ready to run, but is not running because of the GPU Exclusive Application. My PC has 3 GPUs that are not being used by BOINC because of the GPU Exclusive Application, even though I have plenty of GPU work to run on them. Each GPU Work Item is set to use "1 CPU + 1 NVIDIA GPU".
3. (Probably required) For the projects that can possibly get CPU work, set them for Resource Share 0, so they are backup projects that will only get work when resources are idle.
4. (Optional) Set BOINC CPU Usage % to something that you can easily monitor. I have intentionally set mine for 50%, on an 8-core-16-thread PC.
5. BUG: BOINC Only runs 5 CPU threads, and does not fetch work for the other 3 CPUs.

**Expected behavior**
Work Fetch should see nidle of 3, to get work to use the CPUs, when the GPUs are not usable due to GPU Exclusive application in use.

**System Information**
 - OS: Windows 10 x64
 - BOINC Version: 7.16.5 x64

**Additional context**
Here's a work fetch log, when BOINC had the GPU Exclusive Application running and was only using 5 CPUs, despite being configured for 50% to use 8 CPUs .... and work fetch did not get work!

```
5/3/2020 11:40:23 PM |  | choose_project(): 1588563623.786711
5/3/2020 11:40:23 PM |  | [work_fetch] ------- start work fetch state -------
5/3/2020 11:40:23 PM |  | [work_fetch] target work buffer: 172800.00 + 43200.00 sec
5/3/2020 11:40:23 PM |  | [work_fetch] --- project states ---
5/3/2020 11:40:23 PM | https://boinc.drugdiscoveryathome.com/ | [work_fetch] REC 0.000 prio -0.000 can't request work: master URL fetch pending (82483.75 sec)
5/3/2020 11:40:23 PM | Acoustics@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Albert@Home | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Amicable Numbers | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | andersonattack | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Asteroids@home | [work_fetch] REC 12488.524 prio -1000.027 can request work
5/3/2020 11:40:23 PM | BOINC alpha test | [work_fetch] REC 0.000 prio -0.000 can't request work: master URL fetch pending (82491.80 sec)
5/3/2020 11:40:23 PM | BURP | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | CAS@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Citizen Science Grid | [work_fetch] REC 0.000 prio -1000.000 can request work
5/3/2020 11:40:23 PM | climateprediction.net | [work_fetch] REC 1.178 prio -1000.000 can request work
5/3/2020 11:40:23 PM | collatz | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Cosmology@Home | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | DENIS@Home | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Distributed Hardware Evolution Project | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Einstein@Home | [work_fetch] REC 25172.755 prio -1000.055 can request work
5/3/2020 11:40:23 PM | Enigma@Home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Fight Neglected Diseases | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Gerasim@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | GoofyxGrid@Home CPU | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | GoofyxGrid@Home NCI | [work_fetch] REC 0.000 prio 0.000 can't request work: non CPU intensive
5/3/2020 11:40:23 PM | GPUGRID | [work_fetch] REC 415991.382 prio -0.917 can request work
5/3/2020 11:40:23 PM | latinsquares | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | LHC@home | [work_fetch] REC 179.873 prio -1000.000 can request work
5/3/2020 11:40:23 PM | lhcathome-dev | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Milkyway@Home | [work_fetch] REC 2929.037 prio -1000.006 can request work
5/3/2020 11:40:23 PM | MindModeling@Beta | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Moo! Wrapper | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | nanoHUB_at_home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | NFS@Home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | NumberFields@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | odlk | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | primaboinca | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | PrimeGrid | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | QMC@HOME | [work_fetch] REC 0.000 prio -1000.000 can request work
5/3/2020 11:40:23 PM | Quake-Catcher Network | [work_fetch] REC 0.000 prio 0.000 can't request work: non CPU intensive
5/3/2020 11:40:23 PM | Radioactive@Home | [work_fetch] REC 0.000 prio 0.000 can't request work: non CPU intensive
5/3/2020 11:40:23 PM | Rake search of diagonal Latin squares | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | ralph@home | [work_fetch] REC 0.000 prio -1000.000 can request work
5/3/2020 11:40:23 PM | RNA World | [work_fetch] REC 898.733 prio -0.002 can't request work: scheduler RPC backoff (3374.50 sec)
5/3/2020 11:40:23 PM | Rosetta@home | [work_fetch] REC 138.354 prio -1000.000 can request work
5/3/2020 11:40:23 PM | SAT@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | SETI@home | [work_fetch] REC 807.372 prio -1000.002 can request work
5/3/2020 11:40:23 PM | SETI@home Beta Test | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | SRBase | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | TN-Grid Platform | [work_fetch] REC 1524.790 prio -1000.003 can request work
5/3/2020 11:40:23 PM | Universe@Home | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | VGTU project@Home | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | wanless2 | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | World Community Grid | [work_fetch] REC 1514.008 prio -1000.003 can request work
5/3/2020 11:40:23 PM | WUProp@Home | [work_fetch] REC 0.013 prio -0.000 can't request work: non CPU intensive
5/3/2020 11:40:23 PM | XANSONS for COD | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | yafu | [work_fetch] REC 0.000 prio -0.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | yoyo@home | [work_fetch] REC 0.000 prio -1000.000 can't request work: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM |  | [work_fetch] --- state for CPU ---
5/3/2020 11:40:23 PM |  | [work_fetch] shortfall 906791.57 nidle 0.00 saturated 10454.23 busy 0.00
5/3/2020 11:40:23 PM | https://boinc.drugdiscoveryathome.com/ | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Acoustics@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Albert@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Amicable Numbers | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | andersonattack | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Asteroids@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | BOINC alpha test | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | BURP | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | CAS@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Citizen Science Grid | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | climateprediction.net | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | collatz | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Cosmology@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | DENIS@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Distributed Hardware Evolution Project | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Einstein@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Enigma@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Fight Neglected Diseases | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Gerasim@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | GoofyxGrid@Home CPU | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | GPUGRID | [work_fetch] share 0.000 project is backed off  (resource backoff: 1238.76, inc 2400.00)
5/3/2020 11:40:23 PM | latinsquares | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | LHC@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | lhcathome-dev | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Milkyway@Home | [work_fetch] share 0.000 account manager prefs
5/3/2020 11:40:23 PM | MindModeling@Beta | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Moo! Wrapper | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | nanoHUB_at_home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | NFS@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | NumberFields@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | odlk | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | primaboinca | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | PrimeGrid | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | QMC@HOME | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Rake search of diagonal Latin squares | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | ralph@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | RNA World | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Rosetta@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | SAT@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | SETI@home | [work_fetch] share 0.000 account manager prefs
5/3/2020 11:40:23 PM | SETI@home Beta Test | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | SRBase | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | TN-Grid Platform | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Universe@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | VGTU project@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | wanless2 | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | World Community Grid | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | XANSONS for COD | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | yafu | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | yoyo@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM |  | [work_fetch] --- state for NVIDIA GPU ---
5/3/2020 11:40:23 PM |  | [work_fetch] shortfall 97879.74 nidle 0.00 saturated 146698.74 busy 73349.37
5/3/2020 11:40:23 PM | https://boinc.drugdiscoveryathome.com/ | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Acoustics@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Albert@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Amicable Numbers | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | andersonattack | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Asteroids@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | BOINC alpha test | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | BURP | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | CAS@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Citizen Science Grid | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | climateprediction.net | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | collatz | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Cosmology@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | DENIS@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Distributed Hardware Evolution Project | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Einstein@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Enigma@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | Fight Neglected Diseases | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Gerasim@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | GoofyxGrid@Home CPU | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | GPUGRID | [work_fetch] share 0.000 project is backed off  (resource backoff: 3380.26, inc 4800.00)
5/3/2020 11:40:23 PM | latinsquares | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | LHC@home | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | lhcathome-dev | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Milkyway@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | MindModeling@Beta | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | Moo! Wrapper | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | nanoHUB_at_home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | NFS@Home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | NumberFields@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | odlk | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | primaboinca | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | PrimeGrid | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | QMC@HOME | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | Rake search of diagonal Latin squares | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | ralph@home | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | RNA World | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | Rosetta@home | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | SAT@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | SETI@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | SETI@home Beta Test | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | SRBase | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | TN-Grid Platform | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | Universe@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | VGTU project@Home | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | wanless2 | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM | World Community Grid | [work_fetch] share 0.000 no applications
5/3/2020 11:40:23 PM | XANSONS for COD | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | yafu | [work_fetch] share 0.000
5/3/2020 11:40:23 PM | yoyo@home | [work_fetch] share 0.000 zero resource share
5/3/2020 11:40:23 PM |  | [work_fetch] ------- end work fetch state -------
5/3/2020 11:40:23 PM | https://boinc.drugdiscoveryathome.com/ | choose_project: scanning
5/3/2020 11:40:23 PM | https://boinc.drugdiscoveryathome.com/ | skip: master URL fetch pending
5/3/2020 11:40:23 PM | Quake-Catcher Network | choose_project: scanning
5/3/2020 11:40:23 PM | Quake-Catcher Network | skip: non CPU intensive
5/3/2020 11:40:23 PM | Albert@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Albert@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | GoofyxGrid@Home CPU | choose_project: scanning
5/3/2020 11:40:23 PM | GoofyxGrid@Home CPU | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | andersonattack | choose_project: scanning
5/3/2020 11:40:23 PM | andersonattack | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | VGTU project@Home | choose_project: scanning
5/3/2020 11:40:23 PM | VGTU project@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | BOINC alpha test | choose_project: scanning
5/3/2020 11:40:23 PM | BOINC alpha test | skip: master URL fetch pending
5/3/2020 11:40:23 PM | GoofyxGrid@Home NCI | choose_project: scanning
5/3/2020 11:40:23 PM | GoofyxGrid@Home NCI | skip: non CPU intensive
5/3/2020 11:40:23 PM | MindModeling@Beta | choose_project: scanning
5/3/2020 11:40:23 PM | MindModeling@Beta | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | collatz | choose_project: scanning
5/3/2020 11:40:23 PM | collatz | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Cosmology@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Cosmology@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | DENIS@Home | choose_project: scanning
5/3/2020 11:40:23 PM | DENIS@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Radioactive@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Radioactive@Home | skip: non CPU intensive
5/3/2020 11:40:23 PM | Rake search of diagonal Latin squares | choose_project: scanning
5/3/2020 11:40:23 PM | Rake search of diagonal Latin squares | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | lhcathome-dev | choose_project: scanning
5/3/2020 11:40:23 PM | lhcathome-dev | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Fight Neglected Diseases | choose_project: scanning
5/3/2020 11:40:23 PM | Fight Neglected Diseases | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | SETI@home Beta Test | choose_project: scanning
5/3/2020 11:40:23 PM | SETI@home Beta Test | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Universe@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Universe@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | XANSONS for COD | choose_project: scanning
5/3/2020 11:40:23 PM | XANSONS for COD | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | yafu | choose_project: scanning
5/3/2020 11:40:23 PM | yafu | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | WUProp@Home | choose_project: scanning
5/3/2020 11:40:23 PM | WUProp@Home | skip: non CPU intensive
5/3/2020 11:40:23 PM | RNA World | choose_project: scanning
5/3/2020 11:40:23 PM | RNA World | skip: scheduler RPC backoff
5/3/2020 11:40:23 PM | GPUGRID | choose_project: scanning
5/3/2020 11:40:23 PM | GPUGRID | can't fetch CPU: project is backed off
5/3/2020 11:40:23 PM | GPUGRID | can't fetch NVIDIA GPU: project is backed off
5/3/2020 11:40:23 PM | Distributed Hardware Evolution Project | choose_project: scanning
5/3/2020 11:40:23 PM | Distributed Hardware Evolution Project | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Enigma@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Enigma@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Gerasim@home | choose_project: scanning
5/3/2020 11:40:23 PM | Gerasim@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | latinsquares | choose_project: scanning
5/3/2020 11:40:23 PM | latinsquares | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Moo! Wrapper | choose_project: scanning
5/3/2020 11:40:23 PM | Moo! Wrapper | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | nanoHUB_at_home | choose_project: scanning
5/3/2020 11:40:23 PM | nanoHUB_at_home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | NFS@Home | choose_project: scanning
5/3/2020 11:40:23 PM | NFS@Home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | NumberFields@home | choose_project: scanning
5/3/2020 11:40:23 PM | NumberFields@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | odlk | choose_project: scanning
5/3/2020 11:40:23 PM | odlk | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | primaboinca | choose_project: scanning
5/3/2020 11:40:23 PM | primaboinca | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | PrimeGrid | choose_project: scanning
5/3/2020 11:40:23 PM | PrimeGrid | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | QMC@HOME | choose_project: scanning
5/3/2020 11:40:23 PM | QMC@HOME | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | QMC@HOME | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | ralph@home | choose_project: scanning
5/3/2020 11:40:23 PM | ralph@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | ralph@home | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | Acoustics@home | choose_project: scanning
5/3/2020 11:40:23 PM | Acoustics@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | SRBase | choose_project: scanning
5/3/2020 11:40:23 PM | SRBase | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Amicable Numbers | choose_project: scanning
5/3/2020 11:40:23 PM | Amicable Numbers | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | SAT@home | choose_project: scanning
5/3/2020 11:40:23 PM | SAT@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | wanless2 | choose_project: scanning
5/3/2020 11:40:23 PM | wanless2 | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | BURP | choose_project: scanning
5/3/2020 11:40:23 PM | BURP | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | CAS@home | choose_project: scanning
5/3/2020 11:40:23 PM | CAS@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | Citizen Science Grid | choose_project: scanning
5/3/2020 11:40:23 PM | Citizen Science Grid | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Citizen Science Grid | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | yoyo@home | choose_project: scanning
5/3/2020 11:40:23 PM | yoyo@home | skip: "no new tasks" requested via Manager
5/3/2020 11:40:23 PM | climateprediction.net | choose_project: scanning
5/3/2020 11:40:23 PM | climateprediction.net | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | climateprediction.net | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | Rosetta@home | choose_project: scanning
5/3/2020 11:40:23 PM | Rosetta@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Rosetta@home | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | LHC@home | choose_project: scanning
5/3/2020 11:40:23 PM | LHC@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | LHC@home | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | SETI@home | choose_project: scanning
5/3/2020 11:40:23 PM | SETI@home | can't fetch CPU: account manager prefs
5/3/2020 11:40:23 PM | SETI@home | can't fetch NVIDIA GPU: zero resource share
5/3/2020 11:40:23 PM | World Community Grid | choose_project: scanning
5/3/2020 11:40:23 PM | World Community Grid | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | World Community Grid | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | TN-Grid Platform | choose_project: scanning
5/3/2020 11:40:23 PM | TN-Grid Platform | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | TN-Grid Platform | can't fetch NVIDIA GPU: no applications
5/3/2020 11:40:23 PM | Milkyway@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Milkyway@Home | can't fetch CPU: account manager prefs
5/3/2020 11:40:23 PM | Milkyway@Home | can't fetch NVIDIA GPU: zero resource share
5/3/2020 11:40:23 PM | Asteroids@home | choose_project: scanning
5/3/2020 11:40:23 PM | Asteroids@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Asteroids@home | can't fetch NVIDIA GPU: zero resource share
5/3/2020 11:40:23 PM | Einstein@Home | choose_project: scanning
5/3/2020 11:40:23 PM | Einstein@Home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Einstein@Home | can't fetch NVIDIA GPU: zero resource share
5/3/2020 11:40:23 PM |  | [work_fetch] No project chosen for work fetch

```

**Comment by JacobWKlein** [comment]
The 16-thread PC was setup to 50%, expecting "8 max CPUs" (8 CPU threads) to be running.
The cache had some GPU tasks that couldn't run, due to GPU Exclusive application running.
The cache had 5 CPU tasks that were running.
It should have had 8.
3 CPUs were sitting idle.

**The problem is that work fetch didn't see CPUs as idle.**
5/3/2020 11:40:23 PM |  | [work_fetch] --- state for CPU ---
5/3/2020 11:40:23 PM |  | [work_fetch] shortfall 906791.57 nidle 0.00 saturated 10454.23 busy 0.00

**Work fetch should have asked some of the projects for work! The "can't fetch CPU: zero resource share" reason it gave, indicates that it didn't recognize it was in an "idle CPU" state.**
5/3/2020 11:40:23 PM |  | [work_fetch] ------- end work fetch state -------
...
5/3/2020 11:40:23 PM | QMC@HOME | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | ralph@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Citizen Science Grid | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | climateprediction.net | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Rosetta@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | LHC@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | World Community Grid | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | TN-Grid Platform | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Asteroids@home | can't fetch CPU: zero resource share
5/3/2020 11:40:23 PM | Einstein@Home | can't fetch CPU: zero resource share
...
5/3/2020 11:40:23 PM |  | [work_fetch] No project chosen for work fetch

**Comment by davidpanderson** [comment]
#3703 might fix this.  The same problem would occur, BTW, if you manually suspend GPU computing rather than having an exclusive app start.

**Comment by AenBleidd** [comment]
Fixed via #3703
